### DISCUSSION TITLE
How to create an enumeration of all available models?

### DISCUSSION THREAD

**Discussion by zhuravkov** [SeaQL/sea-orm#1912] [Q&A]
Get all entities fn make .table_name() for each

**Comment by tyt2y3** [comment]
There currently isn't. But probably in `prelude.rs`, one can add:

```rust
pub use super::cake::Entity as Cake;
pub use super::cake_filling::Entity as CakeFilling;
pub use super::filling::Entity as Filling;
pub use super::fruit::Entity as Fruit;
pub use super::vendor::Entity as Vendor;

pub fn all_entities() -> Vec<Box<dyn EntityTrait>> { .. }
```

**Reply by zhuravkov** [reply]
This is one of the necessary functions to develop an admin panel, among many attempts I tried something similar using the code and from the answer I get an error
entity/src/prelude.rs
```
use sea_orm::EntityTrait;

pub use super::user::Entity as User;
pub use super::model2::Entity as Model2;

pub fn all_entities()->Vec<Box<dyn EntityTrait>> { 
    // let mut a= Vec::new();
    // a.push(Box::new(User));
    // a.push(Box::new(Model2));
    vec![Box::new(User),Box::new(Model2)]
 }

```
error[E0191]: the value of the associated types `Column` (from trait `sea_orm::EntityTrait`), `Model` (from trait `sea_orm::EntityTrait`), `PrimaryKey` (from trait `sea_orm::EntityTrait`), `Relation` (from trait `sea_orm::EntityTrait`) must be specified
 --> entity/src/prelude.rs:8:36
  |
8 | pub fn all_entities()->Vec<Box<dyn EntityTrait>> { 
  |                                    ^^^^^^^^^^^ help: specify the associated types: `EntityTrait<Model = Type, Column = Type, Relation = Type, PrimaryKey = Type>`

**Reply by tyt2y3** [reply]
Ah right. I don't have a better idea for now.

**Reply by zhuravkov** [reply]
//////////entity/src/prelude.rs////////////
 pub fn all_entities_str()->Vec<&'static str> { 
    let user = User.table_name();
    let model2 = Model2.table_name();
    vec![user,model2]
 }

////////////////api/src/lib.rs//////////////////
    let models = all_entities_str();
    Migrator::up(&conn, None).await.unwrap();
    let state = AppState { conn , models};

////////////////api/src/handlers.rs/////////////////////////    
let models = &state.models;

So far I have done it so that I can draw the current sidebar and get the current urls in href , but NEXT, use ActiveModelTrait dont work *the value of the associated type `Entity` (from trait `ActiveModelTrait`) must be specified.
How can I define a variable that will correspond not to a specific model, but to a type?
```
    let model = req.param::<&str>("model").unwrap();
    let title = model;
    let current = model;

    let qs:????? = match model {
        "user"  => QueryUser::find_all(conn).await.unwrap(),
        "model2"  => QueryModel2::find_all(conn).await.unwrap(),
        _ => Err(())

    };


    };
```
What should I indicate instead ?????

**Reply by tyt2y3** [reply]
IMHO you now wandered into the staticness of Rust preventing you to do what you wanted.
Actually I'd suggest you to try Seaography and access the schema via GraphQL https://github.com/SeaQL/sea-orm/tree/master/examples/seaography_example
